File: go/two-fer/two_fer.go
// Package twofer solves a coding exercise.
package twofer

import (
	"fmt"
)

// ShareWith returns a computed string.
func ShareWith(name string) string {
	if len(name) == 0 {
		name = "you"
	}

	return fmt.Sprintf("One for %s, one for me.", name)
}

File: fsharp/two-fer/TwoFer.fs
module TwoFer

let twoFer (input: string option): string = 
    match input with
    | None -> sprintf "One for %s, one for me." "you"
    | Some(name) -> sprintf "One for %s, one for me." name

let twoFer2 (input: string option): string = 
    match input with
    | None -> "you"
    | Some(name) -> name
    |> sprintf "One for %s, one for me."

let twoFer3 (input: string option): string =
    input
    |> Option.defaultValue "you"
    |> sprintf "One for %s, one for me."
